src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -156,7 +156,11 @@ internal static void CheckSubmissionOptions(CompilationOptions options)
                 throw new ArgumentException(CodeAnalysisResources.InvalidOutputKindForSubmission, nameof(options));
             }
 
-            if (options.CryptoKeyContainer != null || options.CryptoKeyFile != null || options.DelaySign != null || !options.CryptoPublicKey.IsEmpty)
+            if (options.CryptoKeyContainer != null ||
+                options.CryptoKeyFile != null ||
+                options.DelaySign != null ||
+                !options.CryptoPublicKey.IsEmpty ||
+                (options.DelaySign == true && options.PublicSign))
```

Comment:
@jaredpar This is because DelaySign can be specified on the command line and in source as an attribute. We need to differentiate between passing false and not passing an argument on the command line so we can generate a warning between conflicting options.